File: PHP/ValidPalindromeTest.php
<?php
/**
 * @version PHP 7.3.29
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit ValidPalindromeTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("ValidPalindrome.php");

class ValidPalindromeTest extends TestCase {

    public function testValidPalindrome() {
        $obj = new ValidPalindrome();
        //Test 1
        $want = true;
        $got = $obj->isPalindrome( "A man, a plan, a canal: Panama" );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = false;
        $got = $obj->isPalindrome( "race a car" );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = true;
        $got = $obj->isPalindrome( " " );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 4
        $want = true;
        $got = $obj->isPalindrome( "A man, a plan, a canal -- Panama" );
        echo "\nTest 4: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 5
        $want = false;
        $got = $obj->isPalindrome( "0P" );
        echo "\nTest 5: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/ValidPalindrome_test.go
/**
 * @version GO 1.19.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestValidPalindrome
 */
package GO

import (
	"reflect"
	"testing"
)

func TestValidPalindrome(t *testing.T) {
	tables := []struct {
		want bool
		str  string
	}{
		{true, "A man, a plan, a canal: Panama"},
		{false, "race a car"},
		{true, " "},
		{true, "A man, a plan, a canal -- Panama"},
		{false, "0P"},
	}

	for _, table := range tables {
		got := isPalindromeString(table.str)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %t but the return was this: %t to %s", table.want, got, table.str)
		}
	}
}
